### PR TITLE
add deployment controller without any handler functionality

### PR DESCRIPTION
**Pull request by nirvanagit**

### CODE REVIEW

File: pkg/controller/util.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package controller
+
+import (
+	"github.com/istio-ecosystem/admiral-state-syncer/pkg/config"
+
+	k8sAppsV1 "k8s.io/api/apps/v1"
+)
+
+func GetDeploymentOriginalIdentifier(deployment *k8sAppsV1.Deployment) string {
```
Comment (rtay1188) [inline pkg/controller/util.go:11]: Can we use the GetDeploymentGlobalIdentifier so it has swimlane/partition awareness?

File: pkg/config/types.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+type LabelSet struct {
+	DeploymentAnnotation                string
+	SubsetLabel                         string
+	NamespaceSidecarInjectionLabel      string
+	NamespaceSidecarInjectionLabelValue string
+	AdmiralIgnoreLabel                  string
+	PriorityKey                         string
+	WorkloadIdentityKey                 string //Should always be used for both label and annotation (using label as the primary, and falling back to annotation if the label is not found)
```
Comment (rtay1188) [inline pkg/config/types.go:21]: IdentityPartitionKey for swimlane/partition awareness

File: pkg/client/kube_client.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+	argo "github.com/argoproj/argo-rollouts/pkg/client/clientset/versioned"
+	log "github.com/sirupsen/logrus"
+	istio "istio.io/client-go/pkg/clientset/versioned"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type KubeClientLoader struct{}
```
Comment (vrushalijoshi) [inline pkg/client/kube_client.go]: can we change this to private?

File: pkg/controller/deployment.go [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+		} else {
+			dce.Deployments[env] = &DeploymentItem{
+				Status: status,
+			}
+
+			p.cache[dce.Identity] = dce
+			return nil
+		}
+	}
+	return fmt.Errorf(types.LogCacheFormat, "Update", "Deployment",
```
Comment (vrushalijoshi) [inline pkg/controller/deployment.go:135]: do we need env information also in the log?

File: pkg/controller/deployment.go [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+		deployment.Name, deployment.Namespace, "", "nothing to update, deployment not found in cache")
+}
+
+func (p *deploymentCache) GetByIdentity(key string) map[string]*DeploymentItem {
+	defer p.mutex.Unlock()
+	p.mutex.Lock()
+	dce := p.cache[key]
+	if dce != nil {
+		return dce.Deployments
+	} else {
```
Comment (vrushalijoshi) [inline pkg/controller/deployment.go]: if we return at the end of the method, we may not need this else

**Review by vrushalijoshi — COMMENTED** [review]
Some minor suggestions

File: pkg/controller/util.go (hunk 1 above)
Comment (nirvanagit) [reply pkg/controller/util.go:11]: Added

File: pkg/config/types.go (hunk 2 above)
Comment (nirvanagit) [reply pkg/config/types.go:21]: added

File: pkg/controller/deployment.go (hunk 4 above)
Comment (nirvanagit) [reply pkg/controller/deployment.go:135]: added

File: pkg/controller/deployment.go (hunk 5 above)
Comment (nirvanagit) [reply pkg/controller/deployment.go]: Thanks!

**Review by rtay1188 — APPROVED** [review]

**Pull request merged**